daemon/cmd/cells.go

```diff
@@ -91,6 +92,18 @@ var (
 
 		// IPCache, policy.Repository and CachingIdentityAllocator.
 		cell.Provide(newPolicyTrifecta),
+
+		// Conditionally instantiate the Egress Gateway.
+		cell.Config(egressgateway.DefaultConfig),
+		cell.Provide(
+			func(params egressgateway.Params) *egressgateway.Manager {
+				if !option.Config.EnableIPv4EgressGateway {
```

Comment:
if there's not too much churn maybe pull this into the egress config and out of option.config. But this might be what you alluded to re minimal changes; fine in that case

Reply:
yes, this is what i was referring to. I had a chat with gilberto and maybe there is an easy way to work around this, i'll take another look.

Reply:
Preferably I'd of course see these in `EgressGatewayConfig`, but seeing where the enable flag is referenced it'd get painful to do that in this PR since you'd need to plumb the EgressGatewayConfig struct to all these places that have not yet been modularized. Let's keep that out from this PR for the time being. 

As a transitionary thing we could already add EgressGatewayConfig, keep the option.Config.EnableIPv4EgressGateway, remove the flag registration from `daemon/cmd/daemon_main.go` and do it in EgressGatewayConfig.Flags() instead, and finally set `option.Config.EnableIPv4EgressGateway` somewhere... and the last point is where it gets a bit hairy. It can be made to work by having a `cell.Invoke` at the very top in `daemon/cmd/cells.go` which would depend on `*DaemonConfig` and `EgressGatewayConfig` and bridge the two before the "daemonCell" or anyone else gets access to it. I don't like it that much though. I think it'd be preferable to just do it cleanly once when we're far enough that we can do it.

Another thing here is that most likely `EgressGatewayConfig` is a "datapath" configuration, e.g. it affects the operation of datapath (above e.g. pkg/datapath/loader/base.go) more than control-plane and it should likely live under `pkg/datapath` and possibly stay internal to it. Control-plane components that manage the "egress gateway datapath" would then depend on the config either directly, or via some "egress gateway datapath API" to tell them what can and cannot be done. This is maybe less clear with egress gateway, but with datapath components that require probing to figure out the "actual" configuration, it makes sense to not expose the user provided config struct, but rather the configuration that's the result of merging user configuration and probe results. 

tl;dr: let's do nothing in this PR about the config :grin: